core/cas-server-core-events/src/test/java/org/apereo/cas/support/events/CasAuthenticationEventListenerTests.java

```diff
@@ -125,7 +148,9 @@ public void verifyCasAuthenticationPolicyFailureEvent() {
             new DefaultAuthenticationTransaction(CoreAuthenticationTestUtils.getService(),
                 CollectionUtils.wrap(CoreAuthenticationTestUtils.getCredentialsWithSameUsernameAndPassword())),
             CoreAuthenticationTestUtils.getAuthentication());
+        var current = numEventsReceived.get();
         applicationContext.publishEvent(event);
+        sleep(current + 1);
```

Comment:
All these should be removed and replaced with `await`.

Reply:
I wasn't aware of the Awaitility library.. Thank you for pointing that out.

Reply:
I refactored this a bit.  These are calling either publishEventAndWaitToProcess or waitForSpringEventToProcess, and is now using await.